### ISSUE TITLE
Make X-MS-API-ROLE optional

### ISSUE THREAD

**Issue by thomasgauvin** [Azure/data-api-builder#1240]
DAB currently requires developers to provide `'X-MS-API-ROLE' : 'admin'` as a header when making their requests. However, this is not intuitive for new users, and this is inconvenient for all developers. They expect the role to be used to be assumed as the highest priority, especially when they only consider RBAC (for ex, if I have both authenticated & admin, I want DAB to use admin assuming it has higher permissions).

RBAC should not require the specification of `'X-MS-API-ROLE' : 'admin'`. 

I understand that the explanation provided to me was that, while we can determine the 'most permissive role' for RBAC that the user has (according to the permissions in the config), we cannot determine the most permissive policy in case a policy is applied to a role.

One method I propose is that, if policies apply to the specific query the customer made, we respect the order of the permissions as specified in the config file & match the first policy, with the option to override with the `X-MS-API-ROLE` header.

This solution would not remove any current functionality, while leaving the `X-MS-API-ROLE` header reserved for more advanced use cases (policy matching).

**Comment by itpropro** [comment]
Hey @PaoloPia, 
did this plan actually land on some roadmap? I had some issues with the role header and I realized it’s really uncomfortable to implement a complex authorization model, as you would need to replicate the whole logic of your permission model on the client in addition to the server which adds a lot of friction.

**Comment by itpropro** [comment]
Hey @yorek, 
did you already had some time to discuss this internally? Especially in the combination with SWA, this poses a lot of problems in actual production deployments. If there are some additional information you need to go forward with this, I am happy to help out!

**Comment by yorek** [comment]
Hi @itpropro. It was discussed but the work in order to do it was postponed as other request had higher priority. Let me add @JerryNixon to this thread so he can follow up in the next months.

**Comment by itpropro** [comment]
> Hi @itpropro. It was discussed but the work in order to do it was postponed as other request had higher priority. Let me add @JerryNixon to this thread so he can follow up in the next months.

Could you add this item to the roadmap, at least in the backlog so it won't get lost? It is quite essential, as the current state doesn't allow for any kind of real world authentication with Static Web Apps. It would already be enough to be able to hand over multiple/all roles that the user currently has to "unblock" most scenarios.

**Comment by itpropro** [comment]
@yorek any updates on this? This is still blocking the usage if data-api-builder directly as well as in Azure Static Web Apps for any application that needs more than one or two roles.

**Comment by JerryNixon** [comment]
Not exactly what you are asking for, but we are going to address this to simplify clients.

**Comment by itpropro** [comment]
> Not exactly what you are asking for, but we are going to address this to simplify clients.

Thanks for the reply, but we already migrated everything away from data-api-builder for a long time already and trying to avoid it, until it is possible to implement real world authentication and authorization scenarios.
